autometacal/python/metacal.py

```diff
@@ -25,31 +25,48 @@ def generate_mcal_image(gal_image,
       shearing by g, and reconvolution with reconvolution_psf_image.
   
   """
-  #convert and cast
-  gal_image = tf.convert_to_tensor(gal_image, dtype=tf.float32)
-  psf_image = tf.convert_to_tensor(psf_image, dtype=tf.float32)
-  reconvolution_psf_image = tf.convert_to_tensor(reconvolution_psf_image, dtype=tf.float32)
+  #cast stuff as float32 tensors
+  gal_images = tf.convert_to_tensor(gal_images, dtype=tf.float32)  
+  psf_images = tf.convert_to_tensor(psf_images, dtype=tf.float32)  
+  reconvolution_psf_image = tf.convert_to_tensor(reconvolution_psf_image, dtype=tf.float32)  
+  g = tf.convert_to_tensor(g, dtype=tf.float32)  
   
+  #Get batch info
+  batch_size, nx, ny = gal_images.get_shape().as_list()  
+      
+  #add pads in real space
+  padfactor = 3 #total width of image after padding
```

Comment:
hummmm this is a bit dangerous to hardcode these factors here.

And do we actually need that? or is it enough to ask the user to provide larger stamps?

Reply:
I know, but for now, larger pads create memory problems, and this is the minimum size. 
It would work to provide larger stamps, but in real situations, it adds neighbor objects. I think that if we want to do in another way, we can create a new PR that changes that.

Reply:
can you just add the pad factor as an input to the function? We can fix things later but hard coded factors like these are really bad practice, we should  always avoid that

Reply:
Done!